driver/modern_bpf/helpers/interfaces/attached_programs.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2023 The Falco Authors.
+ *
+ * This file is dual licensed under either the MIT or GPL 2. See MIT.txt
+ * or GPL2.txt for full copies of the license.
+ */
+
+#pragma once
+
+#include <helpers/base/maps_getters.h>
+#include <definitions/events_dimensions.h>
+#include <helpers/store/ringbuf_store_params.h>
+
+/* This enum is used to tell if we are considering a syscall or a tracepoint */
+enum intrumentation_type
+{
+	SYSCALL = 0,
+	TRACEPOINT = 1,
+};
+
+
+static __always_inline void send_drop_e_event()
```

Comment:
you may ask why here we use inline functions instead of tail calls like in all other cases... The issue is that this sampling logic should be used by all bpf programs attached directly in the kernel! Right now we are always using tracepoint programs but if in the next future, we will use different instrumentation points like `kprobe`  or `lsm` and this could become an issue because bpf tail calls are not allowed between bpf programs of different types (`kprobe` -> `tracepoint` or vice-versa)